kong/plugins/request-termination/schema.lua

```diff
@@ -21,6 +21,8 @@ return {
           { message = { type = "string" }, },
           { content_type = { type = "string" }, },
           { body = { type = "string" }, },
+          { echo = { type = "boolean", required = true, default = false }, },
+          { echo_trigger = typedefs.header_name }
```

Comment:
How about `echo_trigger_header` to make it a little more descriptive?

Reply:
it triggers on both header as well as query entries by that name.

Reply:
one thing I have been considering; currently the trigger is called `echo_trigger` since it only applies to the echo response. It could be called generically `trigger` and then apply to the entire plugin behaviour.

I think I would prefer that over the current `echo_trigger`. Thoughts? @hbagdi

Reply:
calling it `trigger` would also allow any endpoint to be converted into a healthcheck endpoint, if you configure the probe request to send the `trigger` along

Reply:
`trigger` makes more sense, yes.
Meta: this small feature is actually a preview of what we have been wanting to Kong for years. A fine-grained way to trigger plugin execution. :)
I don't have any further comment on this thread, so resolving it.